core/primitives/inherent_data.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/inherent_data.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::primitives, InherentData::Error, e) {
+  using E = kagome::primitives::InherentData::Error;
+  switch (e) {
+    case E::IDENTIFIER_ALREADY_EXISTS:
+      return "This identifier already exists";
+  }
+  return "Unknow error";
+}
+
+namespace kagome::primitives {
+
+  outcome::result<void> InherentData::putData(InherentIdentifier identifier,
+                                              common::Buffer inherent) {
+    if (data_.find(identifier) == data_.end()) {
```

Comment:
Save a result of `find` intro variable to avoid double get - first with `find`, second with `[]`

Reply:
But here [] occurs only if  the entry is not found in the map, so I don't think that an iterator to the end is gonna make it much faster.

Reply:
Yeah, right